x/dex/contract/execution.go

```diff
@@ -212,8 +201,9 @@ func ExecutePairsInParallel(ctx sdk.Context, contractAddr string, dexkeeper *kee
 		// need to re-throw panic to the top level goroutine
 		panic("panicked during pair execution")
 	}
+	dexkeeper.SetMatchResult(ctx, contractAddr, types.NewMatchResult(orderResults, cancelResults, settlements))
```

Comment:
once the offchain solution is built, can this be removed?

Reply:
@philipsu522 this will actually be what the offchain indexer reads from for now. We can change it to be event-based later but I want to first see what's the performance impact of emitting lots of events from the load test kartik's about to do